src/mango/src/mango_cursor_view.erl

```diff
@@ -224,13 +225,18 @@ composite_prefix([Col | Rest], Ranges) ->
 % Prefix and the FieldRanges. If that is equal, then
 % choose the index with the least number of
 % fields in the index. If we still cannot break the tie,
-% then choose alphabetically based on ddocId.
+% then choose alphabetically based on (dbname, ddocid, view_name).
 % Return the first element's Index and IndexRanges.
 %
 % In the future we can look into doing a cached parallel
 % reduce view read on each index with the ranges to find
 % the one that has the fewest number of rows or something.
-choose_best_index(_DbName, IndexRanges) ->
+-type range() :: {binary(), any(), binary(), any()} | empty.
```

Comment:
Oops I think this is a mistake, I should have caught it during the second review pass. The comparators are atoms not binary:

```
1> is_atom('$lte').
true
```

A binary would look like `<<"$lte">>`

Reply:
All right -- opened https://github.com/apache/couchdb/pull/4429 to address this.